File: echarts/src/main/java/android/cl/com/echarts/MainActivity.java
package android.cl.com.echarts;

import android.app.Activity;
import android.os.Bundle;
import android.view.KeyEvent;
import android.view.View;
import android.webkit.JsResult;
import android.webkit.WebChromeClient;
import android.webkit.WebView;
import android.webkit.WebViewClient;
import android.widget.Button;


public class MainActivity extends Activity implements View.OnClickListener {

	private Button linechart_bt;
	private Button barchart_bt;
	private Button piechart_bt;
	private WebView chartshow_wb;

	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		setContentView(R.layout.content_main);
		initView();
	}

	/**
	 * 初始化页面元素
	 */
	private void initView() {

		findViewById(R.id.linechart_bt).setOnClickListener(this);
		findViewById(R.id.barchart_bt).setOnClickListener(this);
		findViewById(R.id.scatterchart_bt).setOnClickListener(this);
		findViewById(R.id.kchart_bt).setOnClickListener(this);
		findViewById(R.id.piechart_bt).setOnClickListener(this);
		findViewById(R.id.radarchart_bt).setOnClickListener(this);
		findViewById(R.id.chordchart_bt).setOnClickListener(this);
		findViewById(R.id.forcechart_bt).setOnClickListener(this);
		findViewById(R.id.mapchart_bt).setOnClickListener(this);
		findViewById(R.id.gaugechart_bt).setOnClickListener(this);
		findViewById(R.id.funnelchart_bt).setOnClickListener(this);
		chartshow_wb = (WebView) findViewById(R.id.chartshow_wb);
		//进行webwiev的一堆设置
		//开启本地文件读取（默认为true，不设置也可以）
		chartshow_wb.getSettings().setAllowFileAccess(true);
		//设置编码
		chartshow_wb.getSettings().setDefaultTextEncodingName("utf-8");
		// 设置可以支持缩放
		chartshow_wb.getSettings().setSupportZoom(true);
		// 设置出现缩放工具
		chartshow_wb.getSettings().setBuiltInZoomControls(true);
		// 清除浏览器缓存
		chartshow_wb.clearCache(true);
		//开启脚本支持
		chartshow_wb.getSettings().setJavaScriptEnabled(true);
		//  放在 assets目录
		//获取Assets目录下的文件
		chartshow_wb.loadUrl("file:///android_asset/echart/echarts.html");
//        chartshow_wb.loadUrl("file:///android_asset/test/echarts.html");
//        chartshow_wb.loadUrl("file:///android_asset/test/index.html");
//        chartshow_wb.loadUrl("http://www.baidu.com");
		//在当前页面打开链接了
		chartshow_wb.setWebViewClient(new WebViewClient() {
			public boolean shouldOverrideUrlLoading(WebView view, String url) {
				view.loadUrl(url);
				return true;
			}
		});
		//js加上这个就好啦！
		chartshow_wb.setWebChromeClient(new WebChromeClient() {
			@Override
			public boolean onJsAlert(WebView view, String url, String message, JsResult result) {
				return super.onJsAlert(view, url, message, result);
			}
		});

	}

	@Override
	public void onClick(View v) {
		switch (v.getId()) {
			case R.id.linechart_bt:
				chartshow_wb.loadUrl("javascript:createChart('line',[]);");
//                chartshow_wb.loadUrl("javascript:createChart('line',[89,78,77]);");
//                chartshow_wb.loadUrl("javascript:wave();");
				break;
			case R.id.barchart_bt:
				chartshow_wb.loadUrl("javascript:createChart('bar',[]);");
//                chartshow_wb.loadUrl("javascript:createChart('bar',[89,78,77]);");
				break;
			case R.id.scatterchart_bt:
				chartshow_wb.loadUrl("javascript:createChart('scatter',[]);");
				break;
			case R.id.kchart_bt:
				chartshow_wb.loadUrl("javascript:createChart('k',[]);");
				break;
			case R.id.piechart_bt:
				chartshow_wb.loadUrl("javascript:createChart('pie',[]);");
				break;
			case R.id.radarchart_bt:
				chartshow_wb.loadUrl("javascript:createChart('radar',[]);");
				break;
			case R.id.chordchart_bt:
				chartshow_wb.loadUrl("javascript:createChart('chord',[]);");
				break;
			case R.id.forcechart_bt:
				chartshow_wb.loadUrl("javascript:createChart('force',[]);");
				break;
			case R.id.mapchart_bt:
				chartshow_wb.loadUrl("javascript:createChart('map',[]);");
				break;
			case R.id.gaugechart_bt:
				chartshow_wb.loadUrl("javascript:createChart('gauge',[]);");
				break;
			case R.id.funnelchart_bt:
				chartshow_wb.loadUrl("javascript:createChart('funnel',[]);");
				break;
			default:
				break;
		}
	}

	@Override
//设置回退 在页面内回退
//覆盖Activity类的onKeyDown(int keyCoder,KeyEvent event)方法
	public boolean onKeyDown(int keyCode, KeyEvent event) {
		if ((keyCode == KeyEvent.KEYCODE_BACK) && chartshow_wb.canGoBack()) {
			chartshow_wb.goBack(); //goBack()表示返回WebView的上一页面
			return true;
		}
//        finish();//结束退出程序
//        return false;
		return super.onKeyDown(keyCode, event);
	}


}

File: echartsdemo/src/main/java/com/example/echartsdemo/MainActivity.kt
package com.example.echartsdemo

import android.annotation.SuppressLint
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.util.Log
import android.webkit.WebView
import android.webkit.WebViewClient
import com.google.gson.Gson
import kotlinx.android.synthetic.main.activity_main.*

/**
 * <p>作者：黄思程  2018/6/1  11:44
 * <p>邮箱：codinghuang@163.com
 * <p>作用：
 * <p>描述：EChart测试
 */
class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        initViewLine()
        initViewPie()
    }

    private fun loadLine() {
        val labels = listOf("阅读邮件 ", "回复邮件", "网站搜索", "博客线索", "Facebook", "Twitter   ", "Linkedin")
        val readEmail = listOf(1, 1, 4, 6, 4, 2, 4, 4, 5, 5, 3, 3, 1, 4, 5, 5, 5, 3, 3, 3, 2, 2, 3, 4, 4)
        val backEmail = listOf(0, 0, 1, 0, 0, 0, 2, 2, 1, 3, 0, 0, 0, 0, 3, 2, 2, 0, 0, 0, 1, 1, 0, 0, 0)
        val netSearch = listOf(4, 4, 0, 1, 1, 1, 0, 0, 0, 0, 9, 9, 9, 9, 7, 7, 7, 5, 5, 5, 0, 2, 3, 0, 0)
        val blogLead = listOf(3, 3, 5, 5, 7, 7, 7, 7, 7, 7, 6, 3, 2, 0, 0, 0, 1, 1, 1, 8, 8, 7, 5, 5)
        val faceBook = listOf(30, 10, 36, 20, 8, 5, 5, 5, 5, 6, 6, 1, 2, 3, 4, 5, 4, 3, 2, 1, 0, 0, 7, 7, 7)
        val twitter = listOf(6, 8, 30, 20, 10, 20, 23, 21, 7, 23, 12, 11, 30, 6, 6, 9, 9, 9, 8, 1, 1, 1, 0, 5)
        val linkedIn = listOf(5, 5, 6, 4, 5, 5, 6, 7, 8, 8, 9, 9, 5, 5, 5, 5, 7, 12, 6, 6, 6, 6, 6, 2, 2)
        val lineBean = Gson().toJson(LineBean(labels, readEmail, backEmail, netSearch,
                blogLead, faceBook, twitter, linkedIn))

        Log.d("pieData", lineBean)

        webLineChart.loadUrl("javascript:createChart('line','$lineBean');")
    }

    private fun loadPie() {
        val piePercent = listOf(0.5, 0.4, 0.44, 0.24, 0.26, 0.16)
        val pieTitle = listOf("邮件", "电话", "社交媒体", "网站", "博客", "其他")
        val pieBean = Gson().toJson(PieBean(piePercent, pieTitle))

        Log.d("pieData", pieBean)

        webPieChart.loadUrl("javascript:createChart('pie','$pieBean');")
    }

    @SuppressLint("SetJavaScriptEnabled")
    private fun initViewLine() {
        webLineChart.settings.allowFileAccess = true
        webLineChart.settings.defaultTextEncodingName = "utf-8"
        webLineChart.settings.setSupportZoom(true)
        webLineChart.settings.builtInZoomControls = false
        webLineChart.clearCache(true)
        webLineChart.settings.javaScriptEnabled = true
        webLineChart.loadUrl("file:///android_asset/echarts.html")

        webLineChart.webViewClient = object : WebViewClient() {
            override fun shouldOverrideUrlLoading(view: WebView, url: String): Boolean {
                view.loadUrl(url)
                return true
            }

            override fun onPageFinished(view: WebView?, url: String?) {
                super.onPageFinished(view, url)

                loadLine()
            }
        }
    }

    @SuppressLint("SetJavaScriptEnabled")
    private fun initViewPie() {
        webPieChart.settings.allowFileAccess = true
        webPieChart.settings.defaultTextEncodingName = "utf-8"
        webPieChart.settings.setSupportZoom(false)
        webPieChart.settings.builtInZoomControls = false
        webPieChart.clearCache(true)
        webPieChart.settings.javaScriptEnabled = true
        webPieChart.loadUrl("file:///android_asset/echarts.html")

        webPieChart.webViewClient = object : WebViewClient() {
            override fun shouldOverrideUrlLoading(view: WebView, url: String): Boolean {
                view.loadUrl(url)
                return true
            }

            override fun onPageFinished(view: WebView?, url: String?) {
                super.onPageFinished(view, url)

                loadPie()
            }
        }
    }
}
